torch/utils/backend_registration.py

```diff
@@ -23,9 +26,43 @@ def rename_privateuse1_backend(backend_name: str) -> None:
     Example::
 
         >>> # xdoctest: +SKIP("failing")
-        >>> torch.register_privateuse1_backend("foo")
+        >>> torch.utils.register_privateuse1_backend("foo")
         # This will work, assuming that you've implemented the right C++ kernels
         # to implement torch.ones.
         >>> a = torch.ones(2, device="foo")
-        """
+    """
     return _rename_privateuse1_backend(backend_name)
+
+supported_amp_dtype_for_privateuse1 = set()
+
+def set_amp_supported_dtype(dtypes) -> None:
+    r"""
+    set_amp_supported_dtype(dtypes) -> None
+
+    This is a registration API for external backends that would like to register
+    amp supported dtypes for their own device. And you should define your own device module to
+    supported amp  simultaneously.
+
+    Example::
+
+        >>> # xdoctest: +SKIP("failing")
+        >>> torch.utils.register_privateuse1_backend("foo")
+        >>> torch.utils.set_amp_supported_dtype(torch.float16)
+        """
+    global supported_amp_dtype_for_privateuse1
+
+    if isinstance(dtypes, torch.dtype):
+        supported_amp_dtype_for_privateuse1.add(dtypes)
+    elif isinstance(dtypes, (list, tuple)):
+        for dtype in dtypes:
+            set_amp_supported_dtype(dtype)
+    else:
+        raise ValueError("Excepted torch.dtype or list of torch.dtype, but got", dtypes)
+
+def get_amp_supported_dtype() -> List[torch.dtype]:
```

Comment:
Since all of the other autocast-related API's live on the device module (`torch.custom_backend.___`) - can you move this to be an expected method on the module as well? That way all of the AMP-related API's live in the same place for the custom backend.

That will also let us remove `set_amp_supported_dtype()`, which seems like a footgun (we don't want the user to tell the backend which AMP dtypes it supports).

Reply:
> Since all of the other autocast-related API's live on the device module (`torch.custom_backend.___`) - can you move this to be an expected method on the module as well? That way all of the AMP-related API's live in the same place for the custom backend.
> 
> That will also let us remove `set_amp_supported_dtype()`, which seems like a footgun (we don't want the user to tell the backend which AMP dtypes it supports).

ok. It is good advise, I will give these changes according to your comments. @bdhirsh